hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   low at the end of the current sequence.  This potentially
+                   allows for the creation of longer or otherwise more
+                   complex sequences.'''
+          resval: "0x0"
+        },
+        { bits: "25:23",
+          name: "CSNLEAD",
+          desc: '''CS_N Leading Time.  Indicates the number of half sck cycles,
+                   CSNLEAD+1, to leave between the falling edge of cs_n and
+                   the first edge of sck.  Setting this register to zero
+                   corresponds to the minimum delay of one-half sck cycle'''
+          resval: 0
+        },
+        { bits: "22:20",
+          name: "CSNTRAIL"
+          desc: '''CS_N Trailing Time.  Indicates the number of half sck cycles,
+                   CSNTRAIL+1, to leave between last edge of sck and the rising
+                   edge of cs_n. Setting this register to zero corresponds
+                   to the minimum delay of one-half sck cycle.'''
+          resval: 0
+        },
+        { bits: "19:17",
+          name: "CSNIDLE"
+          desc: '''Minimum idle time between commands. Indicates the minimum
+                   number of sck half-cycles to hold cs_n high between commands.
+                   Setting this register to zero creates a minimally-wide CS_N-high
+                   pulse of one-half sck cycle.'''
+          resval: 0
+        },
+        { bits: "15:0",
+          name: "CLKDIV"
+          desc: '''Core clock divider.  Slows down subsequent SPI transactions by a
+                   factor of (CLKDIV+1) relative to the core clock frequency.  The
+                   period of sck, T(sck), then becomes 2*(CLK_DIV+1)*T(core)'''
+          resval: 0
+        },
+      ]
+    },
+    { name: "COMMAND",
+      desc: "Command Register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "3:0",
+          name: "TX1_CNT",
+          desc: '''The number of 1-byte bursts to transmit in this command,
+                   at the single-line data width (8 sck clocks per cycle).
+                   All other data lines are high-Z during this command phase.
+                   This field is ignored when SPEED = 0.
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "12:4",
+          name: "TXN_CNT",
+          desc: '''The number of 1-byte bursts to transmit in this command,
+                   at the full-data width (Determined by the SPEED field).
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "16:13",
+          name: "DUMMY_CNT",
+          desc: '''Controls the number of sck dummy cycles (NOT bytes)
+                   inserted between end of the last TX byte and the beginning of
+                   the first RX byte.   No data is transmitted or received
+                   and all outputs are high-Z during this command phase.  This
+                   field is ignored when SPEED = 0''',
```

Comment:
I assume this is because standard SPI devices never need dummy cycles? Would it make more sense to obey the field regardless of speed, just for uniformity's sake?

Reply:
It's more because standard SPI devices are technically full duplex. 
NOR flashes don't take advantage of this, and have a TX phase and RX phase even for Standard SPI commands.
The "Fast Read" (i.e. High frequency clock) command does require dummy clocks, but there is never a need to tri-state the bus.

However some *other* SPI devices can be transmitting and receiving at the same time (example you can have an ADC which is constantly sampling, taking in new commands as it outputs previous values).  The difference came about because I was unsure how to handle those cases in this framework.

I my latest version I've changed added a "FULLDUPLEX" config flag.  This will make all the commands homogenous unless this bit is set.

Reply:
I'll move that particular change into this PR.

Reply:
Standard Commands are now by default half-duplex, and follow the same conventions as Dual or Quad commands.   (The option for full duplex still exists, and changes the semantics).  Please let me know if you have any comments on the revised text around this point.

Reply:
(Please let me know if there is anything else on this point)

Reply:
Yep, that looks good to me, thanks. (And thanks for explaining about the full- vs. half-duplex distinction)